x-pack/legacy/plugins/apm/server/lib/services/get_services/get_services_items.ts

```diff
@@ -74,7 +75,9 @@ export async function getServicesItems(setup: Setup) {
 
     return {
       serviceName: bucket.key,
-      agentName: idx(bucket, _ => _.agents.buckets[0].key),
+      agentName: idx(bucket, _ => _.agents.buckets[0].key) as
+        | string
+        | undefined,
```

Comment:
Why is this change necessary? Doesn't `idx` always return `T | undefined`?

Reply:
Oh, `terms bucket key type now accounts for the fact it can be both number and string`. That's a little annoying we have to do this everywhere...

Why is this not necessary for `serviceName: bucket.key` ?

Reply:
Yes, we also ran into this issue. We want to start making it easier to build Terms aggregations on dates.